include/decoder/decoder.hh

```diff
@@ -3,12 +3,55 @@
 
 #include "common/common.hh"
 #include "common/inst.hh"
+#include <type_traits>
```

Comment:
nit: Move it up, separately from our include files

Reply:
Fixed